modules/Core/src/main/java/org/terasology/core/world/generator/worldGenerators/PerlinFacetedWorldGenerator.java

```diff
@@ -82,6 +86,7 @@ protected WorldBuilder createWorld() {
                         //.addRasterizer(new GroundRasterizer(blockManager))
                 .addRasterizer(new SolidRasterizer())
                 .addPlugins()
+                .addListeners()
```

Comment:
The WorldBuilder doesn't need to keep track of the order of the listeners relative to anything else, and you can't add individual listeners in the WorldBuilder as non-plugins, because the addListener method doesn't have the right signature for that, so addListeners might as well be integrated into the WorldBuilder constructor rather than being called later.

Reply:
+1 to this, it'd also save the time of going around and updating every worldgen module out there and this seems like a important enough feature to be added by default

Reply:
I've integrated this into the constructor.